### PR TITLE
misc. build cleanup

### PR DESCRIPTION
**Pull request by lyusupov**

### CODE REVIEW

File: src/Adafruit_TinyUSB.h [hunk 1]
```diff
@@ -36,11 +36,19 @@
...
-#include "arduino/msc/Adafruit_USBD_MSC.h"
+#if CFG_TUD_HID
+  #include "arduino/hid/Adafruit_USBD_HID.h"
+#endif
+#if CFG_TUD_MIDI
+  #include "arduino/midi/Adafruit_USBD_MIDI.h"
+#endif
+#if CFG_TUD_MSC
+  #include "arduino/msc/Adafruit_USBD_MSC.h"
+#endif
```
Comment (hathach) [inline src/Adafruit_TinyUSB.h:51]: do you have any particular issues with these include when its class driver are not enabled ?

Comment (lyusupov) [reply src/Adafruit_TinyUSB.h:51]: I had a custom build with 

#define CFG_TUD_HID 0

it was causing a build failure with 'undefined type: hid_report_type_t' message.

The patch above had fixed the build.

Comment (hathach) [reply src/Adafruit_TinyUSB.h:51]: can you provide more info to reproduce:
- your board
- bsp core & library version
- which sketch you are compiling

Comment (lyusupov) [reply src/Adafruit_TinyUSB.h:51]: Sure.

* PCA10056 build target
* Core 1.1.0 with Adafruit_TinyUSB_Arduino 1.4.4
* Adafruit_TinyUSB_Arduino/examples/CDC/cdc_multi

HID component is disabled in Adafruit_TinyUSB_Arduino/src/arduino/ports/nrf/tusb_config_nrf.h as follows:

//------------- CLASS -------------//
#define CFG_TUD_CDC 1
#define CFG_TUD_MSC 1
#define CFG_TUD_HID 0
#define CFG_TUD_MIDI 1
#define CFG_TUD_VENDOR 1

I've got a build error message:

In file included from /home/user/Arduino/hardware/adafruit/nrf52/libraries/Adafruit_TinyUSB_Arduino/src/Adafruit_TinyUSB.h:44,
                 from /home/user/Arduino/hardware/adafruit/nrf52/libraries/Adafruit_TinyUSB_Arduino/examples/CDC/cdc_multi/cdc_multi.ino:20:
/home/user/Arduino/hardware/adafruit/nrf52/libraries/Adafruit_TinyUSB_Arduino/src/arduino/hid/Adafruit_USBD_HID.h:33:45: error: 'hid_report_type_t' has not been declared
   33 |                                             hid_report_type_t report_type,

File: src/Adafruit_TinyUSB.h [hunk 2]
```diff
@@ -36,11 +36,19 @@
...
 #include "arduino/webusb/Adafruit_USBD_WebUSB.h"
```
Comment (hathach) [inline src/Adafruit_TinyUSB.h:52]: could you wrap webusb within the `CFG_TUD_VENDOR` as well for consistency

**Review by hathach — COMMENTED** [review]
great thank you, this PR makes sense. I only have a request to also wrap the webusb within CFG_TUD_VENDOR to make it consistent.

File: src/Adafruit_TinyUSB.h (hunk 2 above)
Comment (lyusupov) [reply src/Adafruit_TinyUSB.h]: Ok.

Done in https://github.com/adafruit/Adafruit_TinyUSB_Arduino/pull/139/commits/92d7cc9b998850138663229db531d1b2c08e765e

**Review by hathach — APPROVED** [review]
look great, thank you for your PR

**Pull request merged**